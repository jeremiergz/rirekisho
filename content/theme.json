{
  "colors": {
    "background": "white",
    "body1": "#262626",
    "body2": "#575757",
    "caption": "#808080",
    "dark": "#212121",
    "error": "#f44336",
    "info": "#2196f3",
    "primary": "#17365c",
    "secondary": "#31859a",
    "subtitle": "#d3d3d3",
    "success": "#4caf50",
    "text": "#262626",
    "white": "#ffffff"
  },
  "fonts": {
    "main": "'Open Sans', sans-serif",
    "title": "'Baloo Bhaina 2', cursive"
  },
  "fontWeights": {
    "bold": 800,
    "bolder": 600,
    "regular": 400
  }
}
