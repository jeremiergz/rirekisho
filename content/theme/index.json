{
  "colors": {
    "dark": {
      "background": "#212121",
      "body1": "#d3d3d3",
      "body2": "#808080",
      "caption": "#d3d3d3",
      "dark": "#212121",
      "error": "#f44336",
      "info": "#2196f3",
      "primary": "#265a99",
      "secondary": "#50b0c7",
      "subtitle": "#575757",
      "success": "#4caf50",
      "text": "#d3d3d3",
      "white": "#ffffff"
    },
    "light": {
      "background": "white",
      "body1": "#262626",
      "body2": "#575757",
      "caption": "#808080",
      "dark": "#212121",
      "error": "#f44336",
      "info": "#2196f3",
      "primary": "#17365c",
      "secondary": "#31859a",
      "subtitle": "#d3d3d3",
      "success": "#4caf50",
      "text": "#262626",
      "white": "#ffffff"
    }
  },
  "fonts": {
    "main": "'Open Sans', sans-serif",
    "title": "'Baloo Bhaina 2', cursive"
  },
  "fontWeights": {
    "bold": 800,
    "bolder": 600,
    "regular": 400
  }
}
