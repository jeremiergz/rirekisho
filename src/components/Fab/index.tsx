import Button from 'components/Button';
import Box, { BoxProps } from 'components/primitives/Box';
import MenuIcon from 'components/svg/Menu';
import React, { useRef, useState } from 'react';

const baseSpacing = '16px';
const positionMapping = {
  'bottom-left': {
    bottom: baseSpacing,
    left: baseSpacing,
  },
  'bottom-right': {
    bottom: baseSpacing,
    right: baseSpacing,
  },
  'top-left': {
    top: baseSpacing,
    left: baseSpacing,
  },
  'top-right': {
    top: baseSpacing,
    right: baseSpacing,
  },
};

const Fab: React.FC<FabProps> = ({ onClick, position, ...rest }) => {
  const buttonRef = useRef();
  const [hovered, setHovered] = useState(false);
  const handleMouseEnter = () => setHovered(true);
  const handleMouseLeave = () => setHovered(false);
  const handleClick = (e: React.MouseEvent) => {
    if (typeof onClick === 'function') onClick(e);
  };
  const positioning = positionMapping[position];
  return (
    <Box height={64} position="fixed" print={false} width={64} zIndex={1} {...positioning} {...rest}>
      <Button
        animation={hovered ? 'none' : 'pulse 1.5s infinite'}
        backgroundColor="secondary"
        borderRadius="50%"
        height="100%"
        onClick={handleClick}
        onMouseEnter={handleMouseEnter}
        onMouseLeave={handleMouseLeave}
        ref={buttonRef}
        variant="cursor-only"
        width="100%"
      >
        <MenuIcon fill="white" height={24} width={24} />
      </Button>
    </Box>
  );
};

Fab.displayName = 'Fab';

export type FabProps = Omit<BoxProps, 'position'> & {
  onClick?: (event: React.MouseEvent) => void;
  position: 'bottom-left' | 'bottom-right' | 'top-left' | 'top-right';
};
export default Fab;
