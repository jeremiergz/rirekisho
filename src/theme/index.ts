const breakpoints = ['320px', '375px', '425px', '768px', '1024px', '1200px', '1440px', '2560px'] as const;
Object.defineProperties(breakpoints, {
  mobileS: { value: breakpoints[0] },
  mobileM: { value: breakpoints[1] },
  mobileL: { value: breakpoints[2] },
  tablet: { value: breakpoints[3] },
  laptopS: { value: breakpoints[4] },
  laptopM: { value: breakpoints[5] },
  laptopL: { value: breakpoints[6] },
  desktop: { value: breakpoints[7] },
});

const colors = {
  background: 'white',
  dark: '#212121',
  error: '#f44336',
  info: '#2196f3',
  primary: '#17365c',
  secondary: '#31859a',
  success: '#4caf50',
  text: '#262626',
  white: '#ffffff',
} as const;

const fonts = {
  main: "'Open Sans', sans-serif",
  title: "'Baloo Bhaina 2', cursive",
} as const;

const fontWeights = {
  bold: 800,
  bolder: 600,
  regular: 400,
} as const;

const theme = {
  breakpoints,
  colors,
  fonts,
  fontWeights,
} as const;

type Theme = typeof theme;

export type { Theme };
export default theme;
