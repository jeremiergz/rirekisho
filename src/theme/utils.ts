/**
 * Shades a color by a percentage and returns changed color as an hexadecimal color.
 *
 * @param color the hexadecimal color to modify
 * @param percent the shade value as a percentage
 */
function shadeColor(color: string, percent: number) {
  const hex = color.replace('#', '');
  const r = Math.round((parseInt(hex.substring(0, 2), 16) * (100 + percent)) / 100);
  const g = Math.round((parseInt(hex.substring(2, 4), 16) * (100 + percent)) / 100);
  const b = Math.round((parseInt(hex.substring(4, 6), 16) * (100 + percent)) / 100);
  const rHex = (r <= 255 ? r : 255).toString(16).padStart(2, '0');
  const gHex = (g <= 255 ? g : 255).toString(16).padStart(2, '0');
  const bHex = (b <= 255 ? b : 255).toString(16).padStart(2, '0');
  return `#${rHex}${gHex}${bHex}`;
}

export { shadeColor };
