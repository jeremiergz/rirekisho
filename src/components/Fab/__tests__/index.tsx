import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import theme from 'theme';
import Fab from '..';

describe('components/Fab Test Suite', () => {
  it('matches snapshot', () => {
    const jsx = (
      <ThemeProvider theme={theme}>
        <Fab position="bottom-right" />
      </ThemeProvider>
    );
    expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
  });
});
