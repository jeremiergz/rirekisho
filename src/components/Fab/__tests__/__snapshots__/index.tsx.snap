// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Fab Test Suite matches snapshot 1`] = `
<div
  className="sc-AxjAm dHfWsI"
  height={64}
  width={64}
>
  <button
    aria-label="Open Menu"
    className="sc-AxjAm sc-AxirZ bydKTu"
    height="100%"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    width="100%"
  >
    <svg
      className="sc-AxiKw jxzaqN"
      fill="white"
      height={24}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
</div>
`;
